sample-apps/go/app/go.mod

```diff
@@ -0,0 +1,3 @@
+module github.com/GoogleCloudPlatform/opentelemetry-operator-sample/sample-apps/go/app
```

Comment:
nit: maybe move this to sample-apps/go/client instead?  Makes it clearer what it does, IMO

Reply:
I did consider that, but the other samples all use `app` for their main code (they are built into the same binary). So, the shared commands to replace things like `%GCP_PROJECT` in k8s manifests expect `app` and `app.yaml`. For consistency when reading through this sample I decided to leave it as app here too

Reply:
sg